docs/backup/new-backup-strategy.md

```diff
@@ -0,0 +1,164 @@
+
+VolumeSnapshot support is only available for CSI drivers. However, not all the CSI drivers support the volume snapshot functionality. You can find a list of all the drivers with the supported functionalities [here](https://kubernetes-csi.github.io/docs/drivers.html).
+
+An example VolumeSnapshot created to take a snapshot from the PVC named `pvc-to-backup`:
+
+> **NOTE:** The PVC to be backed up must be created using CSI-enabled Storage Class.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: volume-snapshot
+spec:
+  volumeSnapshotClassName: csi-snapshot-class
+  source:
+    persistentVolumeClaimName: pvc-to-backup
+```
+
+An example PVC using this snapshot as the data source:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-restored
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: csi-storage-class
+  resources:
+    requests:
+      storage: 10Gi
+  dataSource:
+    name: volume-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+```
+
+You can find more information about VolumeSnapshots [here](https://kubernetes.io/docs/concepts/storage/volume-snapshots/).
+
+Currently none of the cloud providers support this API out-of-the-box yet. You must follow the instructions below to enable this feature for various providers.
+
+### AKS
+
+Minimum Kubernetes version supported is 1.17.
+
+Install CSI driver following [this documentation](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md).
+
+Then, you can follow [this example](https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/snapshot) to see how you can create a VolumeSnapshot.
+
+### GKE
+
+Minimum Kubernetes version supported is 1.14.
+
+Enable the required feature gate on the cluster following [this document](https://cloud.google.com/kubernetes-engine/docs/how-to/gce-pd-csi-driver#enabling_on_a_new_cluster).
+
+Install CSI driver following the instructions [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/snapshots.md#kubernetes-snapshots-user-guide-alpha).
+
+Then, you can follow [this example](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/snapshots.md#snapshot-example) to see how you can create a VolumeSnapshot.
+
+### Gardener Azure
+
+Gardener Azure does not currently support CSI drivers, that's why VolumeSnapshots cannot be used. Its support is planned for Kubernetes 1.19 [#3](https://github.com/gardener/gardener-extension-provider-azure/issues/3).
+
+### Periodic Job for Volume Snapshots
+
+Users can create a Cronjob to take snapshots of the PersistentVolumes periodically.
+
+You can find an example CronJob with the required Service Account and Roles below.
+
+```yaml
+---
+apiVersion: v1
```

Comment:
Maybe move it into the examples repo and link it?

Reply:
Instead, I can add it as a file to assets folder and link it?